website-custom-views/src/content/development/testing.mdx

```diff
@@ -235,3 +235,84 @@ const renderView = (options = {}) => {
   });
 };
 ```
+
+# End-to-End
+
+To complement unit and integration tests, we recommend to also write End-to-End tests using [Cypress](https://www.cypress.io/).
+
+Cypress is a feature-rich and developer friendly tool to write End-to-End tests and it's very easy to use for testing Custom Views.
+
+Assuming you have already [set up and installed Cypress](https://docs.cypress.io/guides/getting-started/installing-cypress), we recommend to install the following packages:
+
+* [`@testing-library/cypress`](https://testing-library.com/docs/cypress-testing-library/intro/): provides commands to select elements using React/Dom Testing Library.
+* `@commercetools-frontend/cypress`: provides commands specific to Custom Views.
+
+Define a `.env` file in the `cypress` folder, containing some of the required environment variables:
+
+```bash title="cypress/.env"
+CYPRESS_LOGIN_USER=""
+CYPRESS_LOGIN_PASSWORD=""
+CYPRESS_PROJECT_KEY=""
+```
+
+<Info>
+
+The `.env` file should be git ignored. On CI, you can define environment variables within the CI job.
+
+</Info>
+
+In the `cypress/plugins/index.js` file, you need to configure the `task` for Custom View and to load the environment variables:
+
+```js title="cypress/plugins/index.js"
+const { customViewConfig } = require('@commercetools-frontend/cypress/task');
+
+module.exports = (on, cypressConfig) => {
+  // Load the config
+  if (!process.env.CI) {
+    const path = require('path');
+    const envPath = path.join(__dirname, '../.env');
+    require('dotenv').config({ path: envPath });
+  }
+
+  on('task', {
+    customViewConfig,
+  });
+
+  return {
+    ...cypressConfig,
+    env: {
+      ...cypressConfig.env,
+      LOGIN_USER: process.env.CYPRESS_LOGIN_USER,
+      LOGIN_PASSWORD: process.env.CYPRESS_LOGIN_PASSWORD,
+      PROJECT_KEY: process.env.CYPRESS_PROJECT_KEY,
```

Comment:
Maybe we can include here the `CYPRESS_PACKAGE_NAME` variable.

Reply:
Right, thanks for noticing 👍